### ISSUE TITLE
Recipes

### ISSUE THREAD

**Issue by thebigjoe1** [davidhealey/waistline#806]
When creating a recipe, the application wants to know the recipe amount and is a required field. In a similar application I was using for creating a meal or recipe It was Auto calculated based on the ingredients and their amounts required for the whole meal/recipe.  Currently so that everything mathematically works out. I have to put in all my ingredients and add them up separately. What I think the field should be instead is what is the serving amount for the recipe so that it can calculate the appropriate macros based on the Total serving size calculated from all of the ingredients relative to the serving size indicated.

**Comment by EmilJunker** [comment]
How would it do that? Some ingredient amounts might be in gram, others in milliliters, ounces, or number of pieces. They can't just be added up. That's why you must specify an amount and a unit for the whole recipe yourself.

**Comment by thebigjoe1** [comment]
You are correct. It can be in multiple units. So I would propose that if we're using a single unit and we ourselves would have to do the conversion of let's say an ounce to a gram. As long as the unit size is the same across the board of the ingredients, I think it would be extremely helpful if it auto calculated. Again, the onus of having to utilize the same unit would fall on the end user and not on the app itself.

**Comment by thebigjoe1** [comment]
> Yeah, I guess it would be possible to have a button in the recipe editor that calculates for you the sum of all ingredient amounts (based purely on the numbers, ignoring any units) so you can use that as the recipe amount if you want.

Yes, I think it would be greatly beneficial if that's something that could be added and greatly appreciated, at the very least by myself.

**Comment by davidhealey** [comment]
I've added the enhancement tag

**Comment by thebigjoe1** [comment]
Thank you again

**Comment by thebigjoe1** [comment]
Any updates on this request? Just curious if we will see implemented anytime soon. Loving the app and added many recipes all with manual addition of ingredient amounts :)

**Comment by EmilJunker** [comment]
This is not a feature that I'd personally have much use for, so it's not high on my priority list.

@thebigjoe1 As is usually the case with open source projects, your best chance of getting a feature implemented is doing it yourself 😉
